### ISSUE TITLE
Are the reported results from the best epoch or the last epoch?

### ISSUE THREAD

**Issue by wwwmj23** [zzzqzhou/RAM-DSIR#7]
Hi,

Thanks for your great work!

Are the reported results from the best epoch or the last epoch? Thank you.

**Comment by zzzqzhou** [comment]
Thanks for reaching out. We choose the last epoch checkpoint for evaluation.

**Comment by wwwmj23** [comment]
Thank you very much for your prompt response. Additionally, I noticed that the validation results of the final epoch during training differ from the values obtained when testing with model_last. Why does this phenomenon occur, and which values should be considered as the final results?
![image](https://github.com/zzzqzhou/RAM-DSIR/assets/56493902/6d212587-c036-4393-93fd-527af2721a67)
![image](https://github.com/zzzqzhou/RAM-DSIR/assets/56493902/6691d2a9-d9b7-47f1-91ca-66b1bb5e871a)

**Comment by zzzqzhou** [comment]
Sorry for the late reply. This is because we do not freeze batch normalization when testing. It is a trick that can improve the final result on test domain. But in training code, we freeze the batch normalization when evaluating. You can also freeze the batch normalization by adding --freeze_bn argument in testing scripts, then the test result will be same as the result from the final epoch during training.
